Add test for tree.py

beckyjackson opened the pull request:
This adds a test to convert the RDFa from `tree.py` to TTL to compare to a known-good TTL file; resolves #8 

You can run this test using `pytest`.

This feels messier than I had hoped... some notes:

1. I changed some things in `tree.py` to make all the RDFa resources IRIs instead of CURIEs (the links still use CURIEs though). The `pyRdfa` package is... messy and doesn't handle CURIEs in any sane way (at least that I could find). If there are any better python packages to parse RDFa, please let me know.

2. The RDFa includes the whole tree structure, so I ignored `rdfs:subClassOf` statements when looking at any triples that have been added.

2. We could use `robot diff` to compare the files, but that would mean the user must have ROBOT installed. The way I set up the comparison of the triples means that the user doesn't need anything other than what's specified in `requirements.txt`. I'm happy to change this though.

3. I've also renamed the `scripts/` directory to `gizmos/` which is setting us up to package on PyPI, but I also wanted to have a nice module name for importing in the test (`import gizmos.tree`).

4. I feel weird about commiting the OBI db file to the repo, but if we don't, then the user also needs `rdftab` to run the tests...

jamesaoverton commented:
Thanks!

1. I really want `rel` and `resource` to use CURIEs. The solution is to add a `prefix` attribute to the `<body>` or `<html>` element, specifying the prefixes like in Turtle. Then the RDFa parser can resolve the CURIEs. There are examples here http://rdfa.info/play/
2. ok
3. ok
4. ok
5. Maybe store a TSV version in the repo, instead of a binary DB, then load it into SQLite? I'm not sure that's better.

Is there a way to specify that these requirements are for development only? I'd prefer that normal users don't need to install these dependencies.

beckyjackson commented:
> The solution is to add a prefix attribute to the <body> or <html> element, specifying the prefixes like in Turtle

Cool, I didn't know about this! I added the prefixes to the HTML and changed the tree script back to CURIEs.

> Is there a way to specify that these requirements are for development only?

Yep. I set this up, just make sure to test using: `python3 setup.py pytest`
